gems/csv/3.3/csv.rbs

```diff
@@ -1700,6 +1700,109 @@ class CSV < Object
+  #     encoding: 'UTF-32BE:UTF-8'
+  #   would read +UTF-32BE+ data from the file
+  #   but transcode it to +UTF-8+ before parsing.
+  #
+  # ---
+  #
+  # These examples assume prior execution of:
+  #   string = "foo,0\nbar,1\nbaz,2\n"
+  #   path = 't.csv'
+  #   File.write(path, string)
+  #
+  #   string_io = StringIO.new
+  #   string_io << "foo,0\nbar,1\nbaz,2\n"
+  #
+  # ---
+  #
+  # With no block given, returns a new \CSV object.
+  #
+  # Create a \CSV object using a file path:
+  #   csv = CSV.open(path)
+  #   csv # => #<CSV io_type:File io_path:"t.csv" encoding:UTF-8 lineno:0 col_sep:"," row_sep:"\n" quote_char:"\"">
+  #
+  # Create a \CSV object using an open \File:
+  #   csv = CSV.open(File.open(path))
+  #   csv # => #<CSV io_type:File io_path:"t.csv" encoding:UTF-8 lineno:0 col_sep:"," row_sep:"\n" quote_char:"\"">
+  #
+  # Create a \CSV object using a \StringIO:
+  #   csv = CSV.open(string_io)
+  #   csv # => #<CSV io_type:StringIO encoding:UTF-8 lineno:0 col_sep:"," row_sep:"\n" quote_char:"\"">
+  # ---
+  #
+  # With a block given, calls the block with the created \CSV object;
+  # returns the block's return value:
+  #
+  # Using a file path:
+  #   csv = CSV.open(path) {|csv| p csv}
+  #   csv # => #<CSV io_type:File io_path:"t.csv" encoding:UTF-8 lineno:0 col_sep:"," row_sep:"\n" quote_char:"\"">
+  # Output:
+  #   #<CSV io_type:File io_path:"t.csv" encoding:UTF-8 lineno:0 col_sep:"," row_sep:"\n" quote_char:"\"">
+  #
+  # Using an open \File:
+  #   csv = CSV.open(File.open(path)) {|csv| p csv}
+  #   csv # => #<CSV io_type:File io_path:"t.csv" encoding:UTF-8 lineno:0 col_sep:"," row_sep:"\n" quote_char:"\"">
+  # Output:
+  #   #<CSV io_type:File io_path:"t.csv" encoding:UTF-8 lineno:0 col_sep:"," row_sep:"\n" quote_char:"\"">
+  #
+  # Using a \StringIO:
+  #   csv = CSV.open(string_io) {|csv| p csv}
+  #   csv # => #<CSV io_type:StringIO encoding:UTF-8 lineno:0 col_sep:"," row_sep:"\n" quote_char:"\"">
+  # Output:
+  #   #<CSV io_type:StringIO encoding:UTF-8 lineno:0 col_sep:"," row_sep:"\n" quote_char:"\"">
+  # ---
+  #
+  # Raises an exception if the argument is not a \String object or \IO object:
+  #   # Raises TypeError (no implicit conversion of Symbol into String)
+  #   CSV.open(:foo)
+  #
+  def self.open: (String | _ToPath | IO filename_or_io,
+                  ?String mode,
+                  ?newline: untyped,
+                  ?invalid: nil | :replace,
+                  ?undef: :replace,
+                  ?replace: String,
+                  ?encoding: Encoding | String,
+                  **untyped options) { (instance) -> void } -> instance
+               | (String | _ToPath | IO filename_or_io,
+                  ?String mode,
+                  ?newline: untyped,
+                  ?invalid: nil | :replace,
+                  ?undef: :replace,
+                  ?replace: String,
+                  ?encoding: Encoding | String,
+                  **untyped options) -> instance
```

Comment:
The block call appears to be just optional.
If so, we can omit the signature as follows.

```suggestion
  def self.open: (String | _ToPath | IO filename_or_io,
                  ?String mode,
                  ?newline: untyped,
                  ?invalid: nil | :replace,
                  ?undef: :replace,
                  ?replace: String,
                  ?encoding: Encoding | String,
                  **untyped options) ?{ (instance) -> void } -> instance
```